displays/cavity_display/backend/fault.py

```diff
@@ -61,11 +61,19 @@ def __init__(
         self.button_text = button_text
         self.button_macro = button_macro
         self.action = action
+        # Storing PV name as a string instead of making a PV obj
+        self.pv: str = pv
+        self._pv_obj: Optional[PV] = None
 
-        self.pv: PV = PV(pv, connection_timeout=PV_TIMEOUT)
+    @property
+    def pv_obj(self) -> PV:
+        if not self._pv_obj:
+            self._pv_obj = PV(self.pv, connection_timeout = PV_TIMEOUT)
+        return self._pv_obj
 
     def is_currently_faulted(self):
-        return self.is_faulted(self.pv)
+        # Changed from self.pv
```

Comment:
I'm not convinced we need these comments but it doesn't really matter. Great job!

Reply:
Honestly we prob dont lol didnt think about that. I can resubmit without them?